### ISSUE TITLE
Conceptual Theory

### ISSUE THREAD

**Issue by jaytimbadia** [pgjones/quart#111]
Flask is not ASGI framework, but it supports async and await keywords in their routes, what does that mean. Will that not make flask an async. Can you compare performance if flask used with async-await keywords and using a new ASGI framework like Quart?

**Comment by pgjones** [comment]
I think I've answered this [here](https://pgjones.dev/blog/flask-async-quart-sync-2019/), let me know if it doesn't answer your question though.

**Comment by jaytimbadia** [comment]
Yeah, I read that and a great read infact, but can you let me know how, if flask is not asynchronous framework, then what is the point of using async keywords? I mean will that improve its performance. 
And on other hand, Quart is Async framework, then why we still need to use async keyword in its code?

May be you explained it in that post, but I am very new to this and didnt know the internal working functions that you mentioned there. Can you if possible please explain in layman terms? 

Also gunicorn is not async server, then also it has async worker why? I am confused, please help me out?

**Comment by shahprog** [comment]
having async not makes it async, but you can have async in your methods even if inside works aren't async. Same as flask you may have async in routes but inside those operations will obviously block
